### PR TITLE
Add template test cases

### PR DESCRIPTION
**Pull request by caomengmeng**
@tingtli @immarvin 
Add 11 template related  test cases:

lsdef_template
lsdef_template_switch_template
lsdef_template_with_invalid_name
mkdef_template_switch_template_without_attribute
mkdef_template_cec_template_without_remainder
mkdef_template_invalid_template
mkdef_node_with_a_node_template
mkdef_template_cec_template_step_by_step
check_mkdef_node_with_template_priority
rmdef_template
chdef_template
Please help to review code, thanks.

### CODE REVIEW

File: xCAT-test/autotest/testcase/lsdef/cases0 [hunk 1]
```diff
@@ -209,3 +209,43 @@ check:rc==0
...
+check:output=~postscripts\=syslog\,remoteshell\,syncfiles 
+check:output=~ip\=OPTIONAL\:the ip address of the switch
+check:output=~mac\=OPTIONAL\:the MAC address of the switch
+check:output=~switchtype\=OPTIONAL\:The type of switch 
+end
+
+start:lsdef_template_with_invalid_name
+os:linux
+description:lsdef --template with invalid template name
+cmd:lsdef --template | grep "test_with_invalid_name"
```
Comment (tingtli) [inline xCAT-test/autotest/testcase/lsdef/cases0]: what does this sentence mean?

**Comment by tingtli** [comment]
@caomengmeng 
some general comments
1. for the lsdef, how about we write the test cases for all the supported type?
2. for mkdef, same with the lsdef, i think we need cover all the template type
3. for the rmdef, we need first create a template, then rmdef? I think 1 case for no template is not enought. You can put rmdef in mkdef cases.

File: xCAT-test/autotest/testcase/mkdef/cases0 [hunk 2]
```diff
@@ -159,4 +159,155 @@ check:output=~not a valid
...
+os:linux
+description:create a node with swtich template without any attributes
+cmd:result=`lsdef | grep  auto_test_node_1`; if [[ $result =~ "auto_test_node_1" ]]; then echo $result; noderm auto_test_node_1; fi
+cmd:mkdef -t node -o auto_test_node_1 --template switch-template
+check:rc==0
+check:output=~1 object definitions have been created or modified.
+cmd:lsdef auto_test_node_1
+check:rc==0
+check:output=~Object name\: auto_test_node_1
+check:output=~groups=switch
```
Comment (zet809) [inline xCAT-test/autotest/testcase/mkdef/cases0]: Pls keep the "=" consistent.

File: xCAT-test/autotest/testcase/lsdef/cases0 (hunk 1 above)
Comment (caomengmeng) [inline xCAT-test/autotest/testcase/lsdef/cases0]: To make sure there isn't this "test_with_invalid_name" template before test below function

**Comment by tingtli** [comment]
@junxiawang Would you please help review. Thanks!

File: xCAT-test/autotest/testcase/lsdef/cases0 (hunk 1 above)
Comment (junxiawang) [inline xCAT-test/autotest/testcase/lsdef/cases0]: please make sure this template should be not exist

File: xCAT-test/autotest/testcase/rmdef/cases0 [hunk 3]
```diff
@@ -128,3 +128,13 @@ cmd:rmdef -t wrongtype -o testnode
...
 end
+
+start:rmdef_template
+os:linux
+description:try to delete a template, then error messages appear 
+cmd:result=`lsdef | grep  switch-template`; if [[ $result =~ "switch-template" ]]; then echo $result; noderm switch-template; fi
+cmd:rmdef switch-template
+check：rc==1
+check:output=~Error\: Could not find an object named \'switch-template\' of type \'node\'
+check:output=~No objects have been removed from the xCAT database.
```
Comment (junxiawang) [inline xCAT-test/autotest/testcase/rmdef/cases0]: please restore the test environment after your test.

File: xCAT-test/autotest/testcase/lsdef/cases0 (hunk 1 above)
Comment (caomengmeng) [inline xCAT-test/autotest/testcase/lsdef/cases0]: Below check:rc==1, this is used to make sure the template doesn't exist

File: xCAT-test/autotest/testcase/rmdef/cases0 (hunk 3 above)
Comment (caomengmeng) [inline xCAT-test/autotest/testcase/rmdef/cases0]: The template would not be deleted, this case is used to check this function.
So no need to restore it.

**Comment by tingtli** [comment]
@junxiawang please merge it if you have no further comments.
@caomengmeng  remember to add it in bundle cases . Thanks!

**Pull request merged**